### PR TITLE
Feature/gov 67 active deactive users

### PR DESCRIPTION
**Pull request by aleciafb**
Changes:
* Activate User and send notification
* Deactivate User and send notification
* Set current user usertype after login (WIP)
* Cleanup notifications
* Fix bug with pagination

Links: 
https://praekelt.atlassian.net/browse/GOV-67
https://praekelt.atlassian.net/browse/GOV-68

### CODE REVIEW

File: src/actions/api/users.js [hunk 1]
```diff
@@ -76,3 +76,33 @@ export const createUser = ({
...
+        await dispatch(
+            apiFetch(`/userprofile/users/${id}/activate/`, {
+                method: 'POST',
+                body: JSON.stringify({
+                    is_active: true
+                })
+            })
+        );
+    } catch (e) {
+        throw new Error(e.message);
```
Comment (aleciafb) [inline src/actions/api/users.js]: Show error message

File: src/actions/api/users.js [hunk 2]
```diff
@@ -76,3 +76,33 @@ export const createUser = ({
         dispatch(alertMessageActions.add(e.message, 'error'));
     }
 };
+
+export const activateUser = id => async dispatch => {
+    try {
+        await dispatch(
+            apiFetch(`/userprofile/users/${id}/activate/`, {
```
Comment (aleciafb) [inline src/actions/api/users.js]: Test if body needed

File: src/actions/users.js [hunk 3]
```diff
@@ -86,3 +84,35 @@ export const fetchPage = () => async (dispatch, getState) => {
...
+};
+
+export const deactivate = userId => async dispatch => {
+    await dispatch(usersApi.deactivateUser(userId));
+    await dispatch(fetchById(userId));
+    dispatch(alertMessageActions.add(`User deactivated`, 'info'));
+};
+
+export const activateAndNotify = user => async dispatch => {
+    console.log(user);
```
Comment (aleciafb) [inline src/actions/users.js]: remove and notifications to notification action

File: src/components/User/index.js [hunk 4]
```diff
@@ -25,14 +33,30 @@ const User = ({ classes, user, userType, ward, jurisdiction }) => (
...
                     }
                     action={
-                        <IconButton>
-                            <EditIcon />
-                        </IconButton>
+                        isAdmin &&
+                        (user.is_active ? (
+                            <Button
+                                color="secondary"
+                                onClick={() => deactivateUser(user)}
```
Comment (aleciafb) [inline src/components/User/index.js]: Shorten name

File: src/components/UsersTable/index.js [hunk 5]
```diff
@@ -80,17 +83,29 @@ const UsersTable = ({
...
-                                            </Button>
-                                        )}
-                                    </TableCell>
+                                    {isAdmin && (
+                                        <TableCell>
+                                            {user.is_active ? (
+                                                <Button
+                                                    color="secondary"
+                                                    onClick={() =>
+                                                        deactivateUser(user)
```
Comment (aleciafb) [inline src/components/UsersTable/index.js]: shorten function name

File: src/containers/UsersTable.js [hunk 6]
```diff
@@ -33,6 +54,13 @@ const mapDispatchToProps = dispatch => ({
         dispatch(
             replace(`/userprofile/users?limit=${limit}&offset=${page * limit}`)
         );
+    },
+    activateUser(user) {
+        console.log(user);
```
Comment (aleciafb) [inline src/containers/UsersTable.js]: remove

File: src/actions/api/users.js [hunk 7]
```diff
@@ -76,3 +76,27 @@ export const createUser = ({
...
+                method: 'POST'
+            })
+        );
+    } catch (e) {
+        dispatch(alertMessageActions.add(e.message, 'error'));
```
Comment (aleciafb) [inline src/actions/api/users.js]: Will be moved to actions layer with all the other alerts

File: src/actions/auth.js [hunk 8]
```diff
@@ -36,6 +37,7 @@ export const login = (username, password) => async dispatch => {
         saveState({
             auth: { token, userId: userId }
         });
+        await dispatch(fetchCurrentUserType());
```
Comment (aleciafb) [inline src/actions/auth.js]: For now, I added it under login - after the token is set. I'm still thinking of moving it to route subscribers

File: src/actions/api/base.js [hunk 9]
```diff
@@ -43,14 +43,21 @@ export const apiFetch = (endpoint, config = {}, setAuthentication) => async (
             headers: addHeaders(token, setAuthentication)
         });
 
-        if (response.status === 403) {
+        if (response.status === 401) {
```
Comment (jwfwessels) [inline src/actions/api/base.js]: Why 401?

File: src/reducers/auth.js [hunk 10]
```diff
@@ -37,6 +37,18 @@ const userId = (state = null, { type, payload, error }) => {
     }
 };
 
+const userType = (state = null, { type, payload, error }) => {
```
Comment (jwfwessels) [inline src/reducers/auth.js]: Not sure about this pattern vs idiomatic Redux pattern...

File: src/actions/api/base.js (hunk 9 above)
Comment (aleciafb) [reply src/actions/api/base.js]: Sorry, I have fixed this to check for 401. I'll send you the update in a bit

**Pull request merged**